File: counting/maximum-population-year.java
class Solution {
    public int maximumPopulation(int[][] logs) {
        int ans = 0;
        int[] arr = new int[101];
        for (int i = 0; i < logs.length; i++) {
            arr[logs[i][0] - 1950]++;
            arr[logs[i][1] - 1950]--;
        }
        for (int i = 1; i < 101; i++) {
            arr[i] += arr[i-1];
            ans = arr[i] > arr[ans] ? i : ans;
        }
        return ans + 1950;
    }
}

File: counting/maximum-population-year.py
class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        year = defaultdict(int)
        ans, max_p, cur = 0, 0, 0
        for l in logs:
            year[l[0]] += 1
            year[l[1]] -= 1
        
        for key in sorted(year.keys()):
            cur += year[key]
            if cur > max_p:
                max_p = cur
                ans = key
        
        return ans

